ngi_pipeline/engines/rna_ngi/launchers.py

```diff
@@ -0,0 +1,142 @@
+    reference_genome=charon_pj.get('reference')
+    fastq_files=[]
+    if reference_genome and reference_genome != 'other':
+        for sample in analysis_object.project:
+            try:
+                charon_reported_status = charon_session.sample_get(analysis_object.project.project_id,
+                                                                   sample).get('analysis_status')
+                # Check Charon to ensure this hasn't already been processed
+                if charon_reported_status == "UNDER_ANALYSIS":
+                    if not analysis_object.restart_running_jobs:
+                        error_text = ('Charon reports seqrun analysis for project "{}" '
+                                      '/ sample "{}" does not need processing (already '
+                                      '"{}")'.format(analysis_object.project, sample, charon_reported_status))
+                        LOG.error(error_text)
+                        if not analysis_object.config.get('quiet'):
+                            mail_analysis(project_name=analysis_object.project.name, sample_name=sample.name,
+                                          engine_name=analysis_module.__name__,
+                                          level="ERROR", info_text=error_text)
+                        continue
+                elif charon_reported_status == "ANALYZED":
+                    if not analysis_object.restart_finished_jobs:
+                        error_text = ('Charon reports seqrun analysis for project "{}" '
+                                      '/ sample "{}" does not need processing (already '
+                                      '"{}")'.format(analysis_object.project, sample, charon_reported_status))
+                        LOG.error(error_text)
+                        if not analysis_object.config.get('quiet') and not analysis_object.config.get('manual'):
+                            mail_analysis(project_name=analysis_object.project.name, sample_name=sample.name,
+                                          engine_name=analysis_module.__name__,
+                                          level="ERROR", info_text=error_text)
+                        continue
+                elif charon_reported_status == "FAILED":
+                    if not analysis_object.restart_failed_jobs:
+                        error_text = ('FAILED:  Project "{}" / sample "{}" Charon reports '
+                                      'FAILURE, manual investigation needed!'.format(analysis_object.project, sample))
+                        LOG.error(error_text)
+                        if not analysis_object.config.get('quiet'):
+                            mail_analysis(project_name=analysis_object.project.name, sample_name=sample.name,
+                                          engine_name=analysis_module.__name__,
+                                          level="ERROR", info_text=error_text)
+                        continue
```

Comment:
I'm not sure I agree. 
While this behaviour is the same for the two engines, it might require tweaking. For instance, I changed how the libprep QC was handled in the RNA engine, but not in the piper engine. 

If you're talking about refactoring the code intra-engine, however, I don't think it can be refactored much more than what it is. What did you have in mind ?